### ISSUE TITLE
height indicator functions should differentiate true 0s from no observation

### ISSUE THREAD

**Issue by annack84** [Landscape-Data-Commons/terradactyl#169]
When running mean_height() with tall=FALSE to calculate the mean height by species, I'm getting 0 values when a species was not observed in the height data at a plot. I think it would be more accurate for the values for species that were not observed to be NA. Changing the values_fill argument in pivot_wider to NA should do the trick.

Line 94 becomes:
summary <- summary %>% tidyr::pivot_wider(names_from = indicator, values_from = mean_height, values_fill = **NA**)

Line 114 becomes:
summary <- summary %>% tidyr::pivot_wider(names_from = Species, values_from = max_height, values_fill = **NA**)

**Comment by nstauffer** [comment]
I believe that this was a deliberate decision when the code was originally written and has been the case for a few years. We'll make sure that it's the intended behavior and if not return NA values.

**Comment by annack84** [comment]
That makes sense, I can see how this is a debatable issue. My argument for filling with NA instead of 0 is that it's easy enough to replace the NA values with replace_na() if one needs them to be treated as 0s, but it's not so simple to reverse engineer the data processing steps to find out which 0s are true 0s (plant was observed at height 0 - happens occasionally, especially on plots with a steep slope) versus which are 0s indicate that the plant was not observed during height measurements.

**Comment by smccord** [comment]
Yeah, this is a grey area. I’m fine with filling NAs for missing species,
since it is possible to have a species present and height recorded as 0. I
definitely see the argument at the species level. It gets murkier at the
functional group level.

I can’t remember how height_calc works for AIM and to what extent we’d have
issues there.  Nelson can you please confirm with Nathan that he is ok with
this?

**Comment by nstauffer** [comment]
I'll check downstream effects and with Nathan before changing any kind of out-of-the-box behavior.

That said, I just pushed changes to the dev branch adding a new argument in `mean_height()` called `missing_fill` that defaults to 0 but can be set to NA. That should maintain the pre-existing behavior but let anyone who'd rather have NA (or, I guess, any other value) in those places instead.
